link-checker.js

```diff
@@ -0,0 +1,186 @@
+const fs = require("fs");
+const path = require("path");
+const childProcess = require("child_process");
+
+const projects = ["apisix-ingress-controller", "apisix", "apisix-dashboard"];
```

Comment:
I notice that we check projects here, so could we remove some logics from`sync-doc` script?

Reply:
Here I use the same document pull code as the `sync-docs.js`.
The behavior of the two scripts in this part of the function is consistent, but because they are not in the same workflow, each script has these codes.

Reply:
if we need to update some logics, we have to update two files? 🤔 it's ok for now, but I'm sure some contributors will miss this point.

Reply:
Create a common file or import one script into another for reuse?

Reply:
ya, I think so 🤔

Reply:
The generic code in the link check module has been splited.(such as document path generation)